core/src/main/java/org/jdeferred/impl/AbstractDeferredManager.java

```diff
@@ -112,6 +114,45 @@
 	}
 
 	@Override
+	public Promise<MultipleResults, OneReject, MasterProgress> when(Collection<?> callableCollection) {
+		assertNotNull(callableCollection);
+		Object[] callables = callableCollection.toArray();
+		assertNotEmpty(callables);
+
+		Promise[] promises = new Promise[callables.length];
+
+		for (int i = 0; i < callables.length; i++) {
+			if (callables[i] instanceof Promise) {
+                promises[i] = when((Promise<?, ?, ?>) callables[i]);
+            } else if (callables[i] instanceof Runnable) {
+				promises[i] = when((Runnable)callables[i]);
+			} else if (callables[i] instanceof Callable) {
+				promises[i] = when((Callable<?>) callables[i]);
+			} else if (callables[i] instanceof DeferredRunnable) {
+				promises[i] = when((DeferredRunnable<?>) callables[i]);
+			} else if (callables[i] instanceof DeferredCallable) {
+				promises[i] = when((DeferredCallable<?,?>) callables[i]);
+			} else if (callables[i] instanceof DeferredFutureTask) {
+				promises[i] = when((DeferredFutureTask<?,?>) callables[i]);
+			} else if (callables[i] instanceof FutureTask) {
+				promises[i] = when((Runnable) callables[i]);
```

Comment:
This cast was giving me trouble. It wouldn't let me do a FutureTask<?>, saying it gets type erased to Runnable.